api/notd/block_processor.py

```diff
@@ -59,12 +67,45 @@ async def get_transfers_in_block(self, blockNumber: int) -> List[RetrievedTokenT
-        logging.info(f'Found {len(events)} events in block #{blockNumber}')
-        allTokenTransfers = []
-        for eventsChunk in list_util.generate_chunks(events, 5):
-            allTokenTransfers += await asyncio.gather(*[self._process_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in eventsChunk])
-        return [tokenTransfer for tokenTransfer in allTokenTransfers if tokenTransfer]
+        #erc721events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc721TansferEventSignatureHash])
+        #logging.info(f'Found {len(erc721events)} events in block #{blockNumber}')
+        erc1155events = await self.ethClient.get_log_entries(startBlockNumber=blockNumber, endBlockNumber=blockNumber, topics=[self.erc1155TansferEventSignatureHash])
+        print(erc1155events)
+        logging.info(f'Found {len(erc1155events)} erc1155events in block #{blockNumber}')
+
+        erc721TokenTransfers = []
+        erc1155TokenTransfers = []
+        #for erc721EventsChunk in list_util.generate_chunks(erc721events, 5):
+        #    erc721TokenTransfers += await asyncio.gather(*[self._process_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc721EventsChunk])
+        for erc1155EventsChunk in list_util.generate_chunks(erc1155events, 5):
+            erc1155TokenTransfers += await asyncio.gather(*[self._process_erc1155_event(event=dict(event), blockNumber=blockNumber, blockHash=blockHash, blockDate=blockDate) for event in erc1155EventsChunk])
+        return [tokenTransfer for tokenTransfer in erc721TokenTransfers or erc1155TokenTransfers if tokenTransfer]
+
+    async def _process_erc1155_event(self, event: LogReceipt, blockNumber: int, blockHash: str, blockDate: datetime.datetime) -> Optional[RetrievedTokenTransfer]:
+        #print(f"transactionHash: {event['transactionHash'].hex()},fromAddress:{ normalize_address(event['topics'][2].hex())}, toAddress: {normalize_address(event['topics'][3].hex())} ,tokenId: { int.from_bytes(bytes(event['topics'][1]), 'big')}")
+        transactionHash = event['transactionHash'].hex()
+        registryAddress = event['address']
+        logging.debug(f'------------- {transactionHash} ------------')
+        if len(event['topics']) < 4:
+            logging.debug('Ignoring event with less than 4 topics')
+            return None
+        #print(f"trnasactionHash: {event['transactionHash'].hex()},toAddress: {event['topics'][3].hex()}")
+        operator = normalize_address(event['topics'][1].hex())
+        fromAddress = normalize_address(event['topics'][2].hex())
+        toAddress = normalize_address(event['topics'][3].hex())
+        tokenId =  int.from_bytes(bytes(event['topics'][1]), 'big')
```

Comment:
This is wrong but the tokenId isnt indexed so i am trying to find out how to get it